packages/itwinui-react/src/core/ComboBox/ComboBoxMenu.tsx

```diff
@@ -49,18 +58,42 @@ const VirtualizedComboBoxMenu = (props: React.ComponentProps<'div'>) => {
     );
   }, [focusedIndex, menuRef]);
 
-  const { outerProps, innerProps, visibleChildren } = useVirtualization({
-    // 'Fool' VirtualScroll by passing length 1
-    // whenever there is no elements, to show empty state message
-    itemsLength: filteredOptions.length || 1,
-    itemRenderer: virtualItemRenderer,
-    scrollToIndex: focusedVisibleIndex,
-  });
+  useLayoutEffect(() => {
+    if (focusedVisibleIndex) {
+      virtualizer.scrollToIndex(focusedVisibleIndex, { align: 'start' });
+    }
+  }, [focusedVisibleIndex, virtualizer]);
+
+  const innerProps = {
+    style: { willChange: 'transform' },
+  } as const;
 
   return (
-    <Box as='div' {...outerProps} {...rest}>
-      <div {...innerProps} ref={innerProps.ref}>
-        {visibleChildren}
+    <Box
+      as='div'
+      style={{
+        minBlockSize: virtualizer.getTotalSize(),
+        minInlineSize: '100%',
+      }}
+      {...rest}
+    >
+      <div {...innerProps}>
```

Comment:
Is this nested div necessary? Previously there were some props coming from the virtualizer, which seems to not be the case anymore

Reply:
It might not be. Wanted to try and the structure as similar as possible to what was already here, but I can try removing this and see if that causes any issues.

Reply:
One more improvement: You can handle these things directly in `virtualItemRenderer`.

Reply:
Yes, you can pass the `virtualItem` to `virtualItemRenderer` and apply all the props there.

Reply:
Done in this [commit](https://github.com/iTwin/iTwinUI/pull/2061/commits/d50f0c5947cfcf5ef00c593d1819692d2f75b98c).